src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -5570,6 +5570,14 @@ ProjectionNames QueryAnalyzer::resolveFunction(QueryTreeNodePtr & node, Identifi
                 column = function_base->getConstantResultForNonConstArguments(argument_columns, result_type);
             }
 
+            if (column->getDataType() != result_type->getColumnType())
```

Comment:
There is a check `if (column ...)` on the next line. So, the assumption that the column can `null` and it's not checked here